src/Auth/ICredentialProvider.php

```diff
@@ -1,17 +0,0 @@
-<?php
-declare(strict_types=1);
-
-namespace Momento\Auth;
-
-interface ICredentialProvider
```

Comment:
why delete the interface?

Reply:
Yeah, we kind of went back and forth on this but I think @cprice404 is right. Keeping the interface around gives us extra flexibility, and that's definitely what we should be returning from the factory methods, @poppoerika.